components/file_attachment_list/file_attachment_list.jsx

```diff
@@ -67,25 +67,28 @@ export default class FileAttachmentList extends React.Component {
 
     render() {
         const {fileInfos, fileCount, compactDisplay} = this.props;
-        const postFiles = [];
-        let sortedFileInfos = [];
 
-        if (fileInfos && fileInfos.length === 1 && compactDisplay === false) {
-            const fileType = getFileType(fileInfos[0].extension);
-
-            if (fileType === FileTypes.IMAGE || fileType === FileTypes.SVG) {
+        if (compactDisplay === false) {
+            if (fileInfos && fileInfos.length === 1) {
+                const fileType = getFileType(fileInfos[0].extension);
+
+                if (fileType === FileTypes.IMAGE || fileType === FileTypes.SVG) {
+                    return (
+                        <SingleImageView
+                            fileInfo={fileInfos[0]}
+                        />
+                    );
+                }
+            } else if (fileCount === 1) {
```

Comment:
Here is the main fix. Just wrap this with `if (compactDisplay === false)` so that single file attachment will render correctly on compact display.